Project1/Map.h

```diff
@@ -0,0 +1,40 @@
+#pragma once
+#include <vector>
+#include "Unit.h"
+#include "GridCell.h"
+#include "CellPath.h"
+#include "GridCell.h"
+#include <map>
+#include <queue>
+/**
+Map class. Holds a Matrix of GridCells and all of the units. Should be initialized at the start of each chapter.
+*/
+using PathMap = std::map<GridCell*, std::pair<int, GridCell*>>;
+using PathQueue = std::priority_queue<std::pair<int, GridCell*>>;
+class Map {
```

Comment:
I think this should be decomposed into at least 2, probably 3, classes. I'll talk to you about this one.